query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresResultAnalyzer.java

```diff
@@ -1,75 +0,0 @@
-package org.hypertrace.core.query.service.postgres;
-
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.util.concurrent.RateLimiter;
-import java.sql.ResultSet;
-import java.sql.ResultSetMetaData;
-import java.sql.SQLException;
-import java.util.HashMap;
-import java.util.LinkedHashSet;
-import java.util.Map;
-import javax.annotation.Nonnull;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-/** Discovers the map attributes indexes from Postgres Result Set */
-class PostgresResultAnalyzer {
-  private static final Logger LOG = LoggerFactory.getLogger(PostgresResultAnalyzer.class);
-
-  /* Stores the Non-Map Attributes logical name to Physical Name index */
-  private final Map<String, Integer> logicalNameToPhysicalNameIndex;
-  private final ResultSet resultSet;
-  private final Map<String, RateLimiter> attributeLogRateLimitter;
-
-  PostgresResultAnalyzer(
-      ResultSet resultSet,
-      LinkedHashSet<String> selectedAttributes,
-      Map<String, Integer> logicalNameToPhysicalNameIndex) {
-    this.logicalNameToPhysicalNameIndex = logicalNameToPhysicalNameIndex;
-    this.resultSet = resultSet;
-    this.attributeLogRateLimitter = new HashMap<>();
-    selectedAttributes.forEach(e -> attributeLogRateLimitter.put(e, RateLimiter.create(0.5)));
-  }
-
-  /** For each selected attributes build the map of logical name to result index. */
-  static PostgresResultAnalyzer create(
-      ResultSet resultSet,
-      LinkedHashSet<String> selectedAttributes,
-      TableDefinition tableDefinition)
-      throws SQLException {
-    Map<String, Integer> physicalNameColIndexMap = new HashMap<>();
-    ResultSetMetaData metaData = resultSet.getMetaData();
```

Comment:
Is our selection list of SQL queries in the same order as `columnMetadataSet` - https://github.com/hypertrace/query-service/blob/56613dafe1bf8eaa6419a26fcaf3f4c7a07755be/query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java#L109 ?

Reply:
Yes, it is. The conversion from query to SQL is in same sequence. Therefore the result set returned contains data in same sequence.

Reply:
> Yes, it is. The conversion from query to SQL is in same sequence. Therefore the result set returned contains data in same sequence.

I see we are using the selection list from the execution context only, so it will be in the same order.
```
Entry<String, Params> sql =
          request2PostgresSqlConverter.toSQL(
              executionContext, request, executionContext.getAllSelections());
```